__tests__/app.test.js

```diff
@@ -69,16 +69,90 @@ describe("GET /api/reviews", () => {
         });
       });
   });
-  test("200: reviews are sorted by newest first", () => {
+  test("200: reviews are sorted by newest first by default", () => {
     return request(app)
       .get("/api/reviews")
+      .expect(200)
       .then(({ body: { reviews } }) => {
-        expect(reviews).toBeSortedBy(reviews.created_at, {
+        expect(reviews).toBeSortedBy("created_at", {
           descending: true,
           coerce: true,
         });
       });
   });
+  describe("Queries", () => {
+    describe("Category", () => {
+      test("200: Filter by specified category", () => {
+        return request(app)
+          .get("/api/reviews?category=social deduction")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toHaveLength(11);
+            reviews.forEach((review) => {
+              expect(review.category).toBe("social deduction");
+            });
+          });
+      });
+      test("200: filtering by a category that has no reviews returns an empty array", () => {
+        return request(app)
+          .get("/api/reviews?category=drinking")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toEqual([]);
+          });
+      });
+      test("200: does not allow SQL injection", () => {
+        return request(app)
+          .get("/api/reviews?category=euro game ;")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toEqual([]);
+          });
+      });
+    });
+    describe("Sort_by", () => {
+      test("200: retrieves all reviews in order of specified column", () => {
+        return request(app)
+          .get("/api/reviews?sort_by=votes")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toBeSorted({ key: "votes", descending: true });
+          });
+      });
+      test("400: sorting by a non existent column returns bad request", () => {
+        return request(app).get("/api/reviews?sort_by=radness").expect(400);
+      });
+      test("400: does not allow SQL injection", () => {
```

Comment:
again I think this test is checking the above, and I would prefer not to say this is the test against SQL injection, just because this might not be the only way that we could be vulnerable to sql injection